contracts/src/StabilityPool.sol

```diff
@@ -629,53 +661,9 @@ contract StabilityPool is LiquityBase, IStabilityPool, IStabilityPoolEvents {
         uint256 secondPortion =
             (epochToScaleToB[snapshots.epoch][snapshots.scale + 1] + secondPortionPending) / SCALE_FACTOR;
 
-        return initialDeposit * (firstPortion + secondPortion) / snapshots.P / DECIMAL_PRECISION;
-    }
-
-    function _getCollGainFromSnapshots(uint256 initialDeposit, Snapshots memory snapshots)
-        internal
-        view
-        returns (uint256)
-    {
-        /*
-        * Grab the sum 'S' from the epoch at which the stake was made. The Coll gain may span up to one scale change.
-        * If it does, the second portion of the Coll gain is scaled by 1e9.
-        * If the gain spans no scale change, the second portion will be 0.
-        */
-        uint128 epochSnapshot = snapshots.epoch;
-        uint128 scaleSnapshot = snapshots.scale;
-        uint256 S_Snapshot = snapshots.S;
-        uint256 P_Snapshot = snapshots.P;
-
-        uint256 firstPortion = epochToScaleToS[epochSnapshot][scaleSnapshot] - S_Snapshot;
-        uint256 secondPortion = epochToScaleToS[epochSnapshot][scaleSnapshot + 1] / SCALE_FACTOR;
-
-        uint256 collGain = initialDeposit * (firstPortion + secondPortion) / P_Snapshot / DECIMAL_PRECISION;
-
-        return collGain;
-    }
-
-    function _getYieldGainFromSnapshots(uint256 initialDeposit, Snapshots memory snapshots)
-        internal
-        view
-        returns (uint256)
-    {
-        /*
-        * Grab the sum 'B' from the epoch at which the stake was made. The Bold gain may span up to one scale change.
-        * If it does, the second portion of the Bold gain is scaled by 1e9.
-        * If the gain spans no scale change, the second portion will be 0.
-        */
-        uint128 epochSnapshot = snapshots.epoch;
-        uint128 scaleSnapshot = snapshots.scale;
-        uint256 B_Snapshot = snapshots.B;
-        uint256 P_Snapshot = snapshots.P;
-
-        uint256 firstPortion = epochToScaleToB[epochSnapshot][scaleSnapshot] - B_Snapshot;
-        uint256 secondPortion = epochToScaleToB[epochSnapshot][scaleSnapshot + 1] / SCALE_FACTOR;
-
-        uint256 yieldGain = initialDeposit * (firstPortion + secondPortion) / P_Snapshot / DECIMAL_PRECISION;
+        uint256 yieldGain = initialDeposit * (firstPortion + secondPortion) / snapshots.P / DECIMAL_PRECISION;
 
-        return yieldGain;
+        return LiquityMath._min(yieldGain, yieldGainsOwed);
```

Comment:
Shouldn't the ceiling be `yieldGainsOwed + pendingSPYield`?

Reply:
Actually, it's more complicated. I think we should only add `pendingSPYield` if `totalBoldDeposits >= DECIMAL_PRECISION`.

Reply:
Probably explains why all the tests are failing.

Reply:
Fixed: 120792df84b4919ed7334ce2a56b8ed5550f8a21